content/hardware/04.pro/boards/portenta-x8/tutorials/02.x8-fundamentals/portenta-x8-fundamentals.md

```diff
@@ -1,85 +1,264 @@
+#### Portenta X8 & Docker Containers
 
-A custom Over-The-Air (OTA) system update mechanism that is based on a client running on target and a robust Cloud server. And they married Docker-compose as a way to deploy a software solution to a target. This is like having an app store for a particular device with the difference that we're not installing an app but a container that may contain a whole distribution or a minimal distribution running only our app or our set of apps.
+The Portenta X8 leverages Docker for containerization. Docker uses **Linux Containers (LXC)** to create lightweight, isolated environments. With Docker, users can access various tools for building, running, and managing containers. These include:
 
-Additionally, they developed the Cloud side as well. You can use what's called FoundriesFactory, a Cloud DevSecOps subscription service to build, test, deploy, and maintain secure, updatable IoT and Edge products. It provides a unique id and automatic builds of the base system and containers for this system in one place. Let's now take a look at the Foundries.io Factory page.
+- **Dockerfile**: A script that defines the instructions to build a container image. This image is a snapshot of the software and its environment.
+
+- **Docker Daemon**: A service running on the host machine that manages container operations, such as starting, stopping, and networking.
+
+- **Docker Swarm and Kubernetes**: Tools for managing containers at scale, allowing robust deployment strategies and automation across multiple systems.
+
+![Docker host](assets/Docker_host.png "Docker host vs client")
+
+#### Portability and Security
+
+Docker containers ensure that applications remain portable. Because the container bundles everything it needs, it can be moved between machines and cloud environments without modification. This portability makes it easy to deploy and scale applications wherever needed.
+
+Furthermore, container isolation protects applications by preventing security threats from spreading between containers or affecting the host system. This ensures a secure, robust operating environment.
+
+![Images and containers](assets/images_containers.png "Images vs containers")
 
 ### Foundries.io Factory
 
-With the help of the Arduino Cloud integration with *Foundries.io*, you can easily create your Factory right from the Arduino Cloud page. You can set your Factory's platform and name. The Portenta X8 will be the platform in this case.
+With Arduino Cloud's integration with *Foundries.io* via the *Portenta X8 Manager*, you can create a **Foundries.io Factory** to manage Portenta X8 devices and containers. A Factory helps manage, organize, and update devices while simplifying container deployment and team management.

```

Comment:
Let's mention again that this is an optional step

Reply:
Updated with additional consideration based on previous queue of updates